lib/LaTeXML/Package/LaTeX.pool.ltxml

```diff
@@ -5967,11 +5967,12 @@ DefMacro('\IfFileExists{}{}{}', sub {
 
 DefMacro('\InputIfFileExists{}{}{}', sub {
     my ($gullet, $file, $if, $else) = @_;
-    my $file_string = ToString(Expand($file));
+    $file = Expand($file);
+    my $file_string = ToString($file);
     if (FindFile($file_string)) {
       DefMacro('\@filef@und', '"' . $file_string . '" ');
-      Input($file_string);
-      return ($if->unlist); }
+      return Tokens($if, T_CS('\@addtofilelist'), T_BEGIN, $file, T_END,
+        T_CS('\@iinput'), T_BEGIN, $file, T_END); }
```

Comment:
Very nice find! (and unfortunate bug)

I'm a tiny bit spooked by using `\@iinput` here, since the 2021 latex.ltx has the flipped dependency:

```tex
\def\@iinput#1{%
  \InputIfFileExists{#1}{}%
  { ... 
```

Reply:
If this is the most sensible setup in LaTeX.pool for now, that's fine - LGTM!

Reply:
Nice! Looks great to me.